Program.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+
+namespace Zadanie1
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            StreamReader file = new StreamReader("in.txt");
+            List<Set> sets = new List<Set>();
+
+            int sets_count = Convert.ToInt32(file.ReadLine());
+            Console.WriteLine($"Ilosc zbiorow:{sets_count}" + Environment.NewLine);
+            
+            string data = file.ReadLine();
```

Comment:
ciachnij odczyt wejścia do oddzielnej metody (np. ParseInput)

jeżeli metoda nie mieści się na jednym ekranie, to jest za długa i trzeba koniecznie podzielić, inaczej nikt tego nie ogarnie oprócz Ciebie i konkretniejszych wymiataczy